lib/console.js

```diff
@@ -35,20 +37,48 @@ function Console(stdout, stderr) {
   }
 }
 
+function write(ignoreErrors, stream, string) {
+  if (!ignoreErrors) return stream.write(string);
+
+  // There may be an error occurring synchronously (e.g. for files or TTYs)
+  // or asynchronously (e.g. pipes), so handle both situations.
+  try {
+    // Add and later remove a noop error handler to catch synchronous errors.
+    stream.once('error', noop);
+
+    stream.write(string, (err) => {
+      // This conditional evaluates to true if and only if there was an error
+      // that was not already emitted (which happens when the _write callback
+      // is invoked asynchronously).
+      if (err && !stream._writableState.errorEmitted) {
+        // If there was an error, it will be emitted on `stream` as
+        // an `error` event. Adding a `once` listener will keep that error
+        // from becoming an uncaught exception, but since the handler is
+        // removed after the event, non-console.* writes won’t be affected.
+        stream.once('error', noop);
+      }
+    });
+  } catch (e) {
+    // Sorry, there’s no proper way to pass along the error here.
+  } finally {
+    stream.removeListener('error', noop);
```

Comment:
No but that’s not a problem. Asynchronous errors are handled from inside the callback.